### ISSUE TITLE
RDM freezes when searching large amounts of (formatted?) data

### ISSUE THREAD

**Issue by Xalaxis** [redis/RedisDesktopManager#4162]
RDM Version: 0.9.3.817
Environment (OS name and version): Windows 10 Version 1803
Redis Server Version: 3.2.100

Steps to reproduce:

1. To recreate my exact environment, create a list with 473 entries, each containing reasonably complex JSON data with three levels of depth.

2. Attempt to search for any string at all

3. RDM freezes after either the first or second character is typed

![ezgif-3-83d49c27c5](https://user-images.githubusercontent.com/12937964/42590970-2de6468a-853d-11e8-8f21-b4bff6190f42.gif)

Expected result: RDM starts searching the content, or displays some kind of progress indicator

Actual Result: RDM freezes (I've not seen an unfreeze)

**Comment by olgalupuleac** [comment]
This will fix the bug, [ValueTabs.qml:69 ](https://github.com/uglide/RedisDesktopManager/blob/0.9/src/qml/value-editor/ValueTabs.qml#L69):
```
-filterString: "*" + table.searchField.text + "*"
+filterString: table.searchField.text
```

The bug isn't related to JSON or list length, it always occurs while searching for a string in a large text.
It happens because `QSortFilterProxyModel` uses `QString::Contains(QRegExp)`. 
If the regular expression has a pattern`"*<some text>*"`, the complexity looks like O(n^2) there n is a length of the text where we want to find a match.
In case of `"<some text>"` the complexity looks like O(n * m) there m is a length of the substring to be searched for.

Still, if the user wants to use a regular expression which contains `*` in a huge text, the problem will occur again.

**Comment by uglide** [comment]
@olgalupuleac Good finding. Could you please submit PR?

**Comment by uglide** [comment]
Fixed in 0.9.5
